pulsar-broker/src/main/java/org/apache/pulsar/broker/intercept/BrokerInterceptor.java

```diff
@@ -64,6 +64,22 @@ default void beforeSendMessage(Subscription subscription,
                                    MessageMetadata msgMetadata) {
     }
 
+    /**
+     * Intercept messages before sending them to the consumers.
+     *
+     * @param subscription pulsar subscription
+     * @param entry entry
+     * @param ackSet entry ack bitset. it is either <tt>null</tt> or an array of long-based bitsets.
+     * @param msgMetadata message metadata. The message metadata will be recycled after this call.
+     * @param consumer consumer. Consumer which entry are sent to.
+     */
+    default void beforeSendMessage(Subscription subscription,
```

Comment:
There are two methods named `beforeSendMessage`. What happens when user implemented both of them?

Reply:
> There are two methods named beforeSendMessage. What happens when user implemented both of them?

If both of these two methods are implemented, only the new one takes effect since in the `AbastractDispatcher` invokes the new method `interceptor.beforeSendMessage(subscription, entry, ackSet, msgMetadata, consumer)`

Also, this is compatible with the original method without the consumer parameter.

Reply:
@aloyszhang So should we just deprecate the old `beforeSendMessage` method? For less confusion when implementing new `BrokerInterceptor`.

Reply:
Make sense to me.